runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7969,6 +7969,8 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
          uintptr_t protectedResult = 0;
          if (entry->getMethodDetails().isJitDumpMethod())
             {
+            TR::CompilationInfoPerThreadBase::UninterruptibleOperation jitDumpRecompilation(*this);
```

Comment:
This is more of a high level question: In JitDump.cpp we enter an interruptible operation region before we queue a compilation of the method specified on the jitdump option and before a recompilation if we're a crashed comp thread, but not if we're an app thread. This seems to only be necessary for the app thread case, so why not have this over there, or maybe just earlier on in `runJitdump` so that all three calls to `recompileMethodForLog` are covered by the same region?

Reply:
For application thread crashes it is the application thread itself which will run the code in JitDump.cpp in the function `runJitDump`. The (crashed) application thread queues the compilations, and the diagnostic thread is the one that will pick them up. We want to make compilations on the diagnostic thread uninterruptible. This is what the line being reviewed here is doing.

The reason we have `UninterruptibleOperation` objects in JitDump.cpp paths for crashed compilation threads is because I'm rather paranoid that _something_ on the way to queue the compile will cause an interruption to happen. I don't have proof for this, but we do acquire VM access in many places on the path to queuing the compilation. And _some_ of the APIs for acquiring VM access check for interruption. Here is an example in `acquireVMaccessIfNeeded`:

https://github.com/eclipse/openj9/blob/c5b05ac868d5304954cd90df4a61c3fd7044c5d6/runtime/compiler/env/VMJ9.cpp#L359-L376

If you are reasonably certain we don't have a call to `compilationShouldBeInterrupted()` on the path from JitDump.cpp to queuing the compile then yes we can remove the two uses of `UninterruptibleOperation` in JitDump.cpp and leave just the one here being reviewed. This is my preference as well.

Let me know what we should do. Thanks!

Reply:
I seem; no I agree with you, better to err on the side of caution.